src/app.js

```diff
@@ -32,6 +32,24 @@ const app = express();
 // gzip middleware
 app.use(compression());
 
+//If this isn't a local developer environment, improve security by only allowing HTTPS
+if (process.env?.NODE_ENV !== 'development') {
+  app.use((req, res, next) => {
+    //Redirect to HTTPS for security, assuming the AWS ALB isn't forwarding the HTTPS request along
+    if (req.headers['x-forwarded-proto'] !== 'https') {
+      /* Use DOMAIN environment variable instead of relying on req.headers.host.
+       * That protects us from uncontrolled redirects to another domain, which is a type of
+       * HTTP Host header attack (see https://portswigger.net/web-security/host-header#how-to-prevent-http-host-header-attacks)
+       */
+      res.redirect('https://' + process.env.DOMAIN + req.url);
+    } else {
+      //If the request has good security, tell the browser to automatically use HTTPS in the future (for next 1 year)
+      res.setHeader('Strict-Transport-Security', 'max-age=31536000');
```

Comment:
I have to assume the AWS ALB will ignore the Strict-Transport-Security header, since a) Strict-Transport-Security is important to browsers, and b) it's HTTPS listener sending requests to HTTP port is coded into its settings.